README.md

```diff
@@ -51,89 +51,99 @@ rewritePattern('foo.bar', 'u');
+  rewritePattern('.', 's', {
+    dotAllFlag: 'transform'
+  });
+  // → '[\\0-\\uFFFF]'
 
-```js
-rewritePattern('(?<=.)a', '', {
-  'lookbehind': true
-});
-// → '(?<=[\\0-\\t\\x0B\\f\\x0E-\\u2027\\u202A-\\uFFFF])a'
-```
+  rewritePattern('.', 'su', {
+    dotAllFlag: 'transform'
+  });
+  // → '(?:[\\0-\\uD7FF\\uE000-\\uFFFF]|[\\uD800-\\uDBFF][\\uDC00-\\uDFFF]|[\\uD800-\\uDBFF](?![\\uDC00-\\uDFFF])|(?:[^\\uD800-\\uDBFF]|^)[\\uDC00-\\uDFFF])'
+  ```
 
-#### `namedGroup` (default: `false`)
+- `unicodePropertyEscapes` - [Unicode property escapes](property-escapes.md).
 
-Setting this option to `true` enables support for [named capture groups](https://github.com/tc39/proposal-regexp-named-groups).
+  By default they are compiled to Unicode code point escapes of the form `\u{...}`. If the `unicodeFlag` option is set to `'transform'` they often result in larger output, although there are cases (such as `\p{Lu}`) where it actually _decreases_ the output size.
 
-```js
-rewritePattern('(?<name>.)\k<name>', '', {
-  'namedGroup': true
-});
-// → '(.)\1'
-```
+  ```js
+  rewritePattern('\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
+    unicodePropertyEscapes: 'transform'
+  });
+  // → '[\\u{14400}-\\u{14646}]'
 
-#### `onNamedGroup`
+  rewritePattern('\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
+    unicodeFlag: 'transform',
+    unicodePropertyEscapes: 'transform'
+  });
+  // → '(?:\\uD811[\\uDC00-\\uDE46])'
+  ```
 
-This option is a function that gets called when a named capture group is found. It receives two parameters:
-the name of the group, and its index.
+- `namedGroups` - [Named capture groups](https://github.com/tc39/proposal-regexp-named-groups).
 
-```js
-rewritePattern('(?<name>.)\k<name>', '', {
-  'namedGroup': true,
-  onNamedGroup(name, index) {
-    console.log(name, index);
-    // → 'name', 1
-  }
-});
-```
+  ```js
+  rewritePattern('(?<name>.)\\k<name>', '', {
+    namedGroup: "transform"
+  });
+  // → '(.)\1'
+  ```
 
-#### `useUnicodeFlag` (default: `false`)
+<!--
 
-Setting this option to `true` enables the use of Unicode code point escapes of the form `\u{…}`. Note that in regular expressions, such escape sequences only work correctly when the ES2015 `u` flag is set. Enabling this setting often results in more compact output, although there are cases (such as `\p{Lu}`) where it actually _increases_ the output size.
+#### Experimental regular expression features
 
-```js
-rewritePattern('\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
-  'unicodePropertyEscape': true,
-  'useUnicodeFlag': true
-});
-// → '[\\u{14400}-\\u{14646}]'
-```
+These options can be set to `false`, `'parse'` and `'transform'`. When using `'transform'`, the corresponding features are compiled to older syntax that can run in older browsers. When using `'parse'`, they are parsed and left as-is in the output pattern. When using `false` (the default), they result in a syntax error if used.
```

Comment:
It will become a breaking change when an experimental feature becomes stable. For example, let's say we have an imaginative option `useSetNotation: boolean | 'parse' | 'transform'` for [RegExp Set Notation](https://github.com/tc39/proposal-regexp-set-notation). Now if the proposal advanced to stage 4, `regexpu-core` will no longer throw for set notation syntax when people are using `useSetNotation: false`, because stable syntax are always parsed. But they may interpret `useSetNotation: false` as I want to target to a _specific_ ecmascript version which does not support set notation and thus I disable the syntax. In this case I think it'd better to limit options to only 1) `'parse'` and `'transform'` or 2) `false` and `transform`, and when an experimental option is absent, we should not parse them at all. This approach is exactly how Babel handles features: when an experimental parser plugin is absent we assume users don't want to parse them at all. And when a feature materializes, the parser plugin becomes no-op.

Reply:
On the other hand, in Babel you can explicitly disable syntax plugins that are enabled somewhere else:
```js
{
  "plugins": ["@babel/plugin-syntax-class-static-block"],
  "env": {
    "test": {
      "plugins": [
        ["@babel/plugin-syntax-class-static-block", false]
      ]
    }
  }
}
```

When running this with `NODE_ENV=test`, it will stop throwing in as soon as Babel enables parsing for static blocks by default.

In general, relaxing an error is never considered to be a breaking change. We might change the wording to be clear, but I think that the behavior is fine.

Reply:
I don't have a strong opinion here. If it helps, I gotta admit that my initial reaction to the current three-value option was one of slight confusion -- I would have expected a simple binary switch (`true`/`false`) instead -- but then it made sense when I read the docs in this patch.